guides/keras_nlp/quick_tour.py

```diff
@@ -0,0 +1,610 @@
+`SentencePiece`.
+
+In this workflow we train a custom tokenizer on the IMDB data and design a backbone with
+custom transformer architecture. For simplicity we then train directly on the
+classification task. Interested in more details? We wrote an entire guide to pretraining
+and finetuning a custom transformer:
+https://keras.io/guides/keras_nlp/transformer_pretraining/
+"""
+
+"""
+## Train custom vocabulary from IMBD data
+"""
+
+vocab = keras_nlp.tokenizers.compute_word_piece_vocabulary(
+    imdb_train.map(lambda x, y: x),
+    vocabulary_size=10_000,  # Increase to 20_000 for better performance
+    lowercase=True,
+    strip_accents=True,
+    reserved_tokens=["[PAD]", "[START]", "[END]", "[MASK]", "[UNK]"],
+)
+tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(
+    vocabulary=vocab,
+    lowercase=True,
+    strip_accents=True,
+    oov_token="[UNK]",
+)
+
+"""
+
+## Preprocess data with custom tokenizer
+"""
+
+packer = keras_nlp.layers.StartEndPacker(
+    start_value=tokenizer.token_to_id("[START]"),
+    end_value=tokenizer.token_to_id("[END]"),
+    pad_value=tokenizer.token_to_id("[PAD]"),
+    sequence_length=64,
+)
+
+
+def preprocess(x, y):
+    token_ids = packer(tokenizer(x))
+    x = {
+        "token_ids": token_ids,
+        "padding_mask": token_ids != tokenizer.token_to_id("[PAD]"),
+    }
+    return x, y
+
+
+imdb_preproc_train_ds = imdb_train.map(
+    preprocess, num_parallel_calls=tf.data.AUTOTUNE
+).prefetch(tf.data.AUTOTUNE)
+imdb_preproc_val_ds = imdb_test.map(
+    preprocess, num_parallel_calls=tf.data.AUTOTUNE
+).prefetch(tf.data.AUTOTUNE)
+
+imdb_preproc_train_ds.unbatch().take(1).get_single_element()
+
+"""
+
+## Design a tiny transformer
+"""
+
+token_id_input = keras.Input(
+    shape=(None,),
+    dtype="int32",
+    name="token_ids",
+)
+padding_mask = keras.Input(
+    shape=(None,),
+    dtype="int32",
+    name="padding_mask",
+)
+outputs = keras_nlp.layers.TokenAndPositionEmbedding(
+    vocabulary_size=len(vocab),
+    sequence_length=packer.sequence_length,
+    embedding_dim=64,
+)(token_id_input)
+outputs = keras_nlp.layers.TransformerEncoder(
+    num_heads=2,
+    intermediate_dim=128,
+    dropout=0.1,
+)(outputs, padding_mask=padding_mask)
+# Use "[START]" token to classify
+outputs = keras.layers.Dense(2)(outputs[:, 0, :])
+model = keras.Model(
+    inputs={
+        "token_ids": token_id_input,
+        "padding_mask": padding_mask,
+    },
+    outputs=outputs,
+)
+
+model.summary()
+
+"""
+## Train the transformer directly on the classification objective
+"""
+
+model.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=keras.optimizers.experimental.AdamW(5e-5),
+    metrics=keras.metrics.SparseCategoricalAccuracy(),
+    jit_compile=True,
+)
+model.fit(
+    imdb_preproc_train_ds,
+    validation_data=imdb_preproc_val_ds,
+    epochs=3,
+)
+
+"""
+While our classification accuracy is a fairly poor 0.76, the transformer architecture is
```

Comment:
With a tiny Transformer on IMDB you should be able to do close to 0.9. The (basically untuned) hyperparameters in my book (ch 11) will get you to 0.875.

Reply:
Thanks, updated! I had been optimizing for runtime, not performance. Increasing `sequence_length` from 64 to 512 matches your performance.